great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2730,8 +2737,10 @@ def _init_datasources(self) -> None:
                 config_dict = dict(datasourceConfigSchema.dump(config))
                 datasource_config = datasourceConfigSchema.load(config_dict)
                 datasource_config.name = datasource_name
+                # Our overall config has already undergone variable substituted here so
+                # there is no difference between raw_config and substituted_config arg values
                 datasource = self._instantiate_datasource_from_config(
-                    config=datasource_config
+                    raw_config=datasource_config, substituted_config=datasource_config
```

Comment:
I have a feeling this may need to change - if we look above, we substitute our ENTIRE project config (datasources and all). That means at this point, we don't have access to our raw configs.

This is used to instantiate datasources and populate the cache but might this cause issues down the line? Say I want to update an existing datasource in the cache and persist it; will I possibly leak credentials here?

I think I'll address this in a follow-up PR - the nature of the PR will be ensuring we have both `raw_config` and `substituted_config` here so we avoid problems downstream.

Reply:
Or maybe I do that in this PR idk

Reply:
just did it here nevermind!